static/src/javascripts/projects/membership/stripe.js

```diff
@@ -0,0 +1,128 @@
+/* global StripeCheckout */
+define(['common/utils/$', 'bean', 'common/utils/ajax','common/utils/config'], function ($, bean, ajax, config) {
+
+    var checkoutHandler = StripeCheckout.configure({
+        key: config.page.stripePublicToken,
+        locale: 'auto',
+        name: 'The Guardian',
+        allowRememberMe: false
+    });
+
+
+    /* Renders the card details
+     *  card:
+     *   -last4: string of last 4 digits of cc
+     *   -type:  string of card type (capitalised)
+     *   */
+
+    function display(parent, card) {
+        var $parent = $(parent);
+        var $number = $('.js-manage-account-card',$parent);
+        var $last4 = $('.js-manage-account-card-last4', $parent);
+        var $type = $('.js-manage-account-card-type', $parent);
+        var $button = $('.js-manage-account-change-card', $parent);
+        var $updating = $('.js-updating', $parent);
+
+        var oldCardType = $type.data('type');
+        var newCardType = 'i-' + card.type.toLowerCase().replace(' ', '-');
+
+        if (oldCardType) {
+            $type.removeClass(oldCardType);
+        }
+
+
+        $last4.text(card.last4);
+        loading(false);
+        $type.addClass(newCardType);
+        $type.data(newCardType);
+        $parent.removeClass('is-hidden');
+        bean.off($button[0], 'click');
+        bean.on($button[0], 'click', handler($parent));
+
+
+        /*
+         * Closes over the event handler for the Change Card button
+         */
+        function handler() {
+
+            var product = $parent.data('product');
+            var endpoint = config.page.userAttributesApiUrl + '/me/' + product + '-update-card';
+            var email = $button.data('email');
+            return function (e) {
+                e.preventDefault();
+                loading(true);
+                checkoutHandler.open({
+                    email: email,
+                    description: 'Update your card details',
+                    panelLabel: 'Update',
+                    token: update(endpoint),
+                    closed: function(){
+                        loading(false);
+                    }
+
+                });
+                // Close Checkout on page navigation:
+                window.addEventListener('popstate', function() {
+                    handler.close();
```

Comment:
The `handler()` function doesn't have a `close` method. Should this be `checkoutHandler.close()`